src/main/java/seedu/address/logic/commands/MatchBuyerCommand.java

```diff
@@ -0,0 +1,99 @@
+            + "Pass in " + PREFIX_STRICT + " after the index for a stricter but possibly less matches."
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_STRICT;
+
+    public static final String MESSAGE_MATCHED_BUYER_SUCCESS = "%s matched properties for the buyer:\n%s";
+
+    private final Index targetIndex;
+    private final boolean isMatchingAll;
+
+    /**
+     * Constructor for MatchBuyerCommand.
+     *
+     * @param targetIndex the index of the property to be matched.
+     * @param isMatchingAll whether all the conditions specified must be satisfied in each of the resulting buyers.
+     */
+    public MatchBuyerCommand(Index targetIndex, boolean isMatchingAll) {
+        this.targetIndex = targetIndex;
+        this.isMatchingAll = isMatchingAll;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Buyer> lastShownList = model.getFilteredPersonList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_BUYER_DISPLAYED_INDEX);
+        }
+
+        Buyer buyerToMatch = lastShownList.get(targetIndex.getZeroBased());
+
+        ArrayList<Predicate<Property>> predicatesList = new ArrayList<>();
+
+        if (buyerToMatch.getDesiredCharacteristics().isEmpty() && buyerToMatch.getPriceRange().isEmpty()) {
+            throw new
+                    CommandException("Cannot match Buyer who has no desired characteristics or price range indicated!");
+        } else {
+            if (buyerToMatch.getPriceRange().isPresent()) {
+                predicatesList.add(new FilterPropsByPricePredicate(buyerToMatch.getPriceRange().get()));
+            }
+
+            if (buyerToMatch.getDesiredCharacteristics().isPresent()) {
+                predicatesList.add(new FilterPropsContainingAnyCharacteristicPredicate(
+                        buyerToMatch.getDesiredCharacteristics().get()));
+            }
+        }
+
+        assert(!predicatesList.isEmpty());
+
+        Predicate<Property> combinedPredicate;
+        if (isMatchingAll) {
+            combinedPredicate = predicatesList.stream().reduce(Predicate::and).get();
+        } else {
+            combinedPredicate = predicatesList.stream().reduce(Predicate::or).get();
+        }
+
+        new FilterPropertiesCommand(combinedPredicate).execute(model);
+
+        return new CommandResult(String.format(MESSAGE_MATCHED_BUYER_SUCCESS,
+                model.getFilteredPropertyList().size(), buyerToMatch));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof MatchBuyerCommand // instanceof handles nulls
+                && targetIndex.equals(((MatchBuyerCommand) other).targetIndex)); // state check
+    }
+}
```

Comment:
I just realised that we should also check `isMatchingAll` in the equals method as well

Same for matchprop